wamp/peer.go

```diff
@@ -1,64 +1,37 @@
 package wamp
 
 import (
-	"context"
 	"errors"
 	"time"
 )
 
 // Peer is the interface implemented by endpoints communicating via WAMP.
 type Peer interface {
-	// Sends the message to the peer.
-	Send(Message) error
-
-	// SendCtx sends the message to the peer, and uses a context to cancel or
-	// timeout sending the message when blocked waiting to write to the peer.
-	SendCtx(context.Context, Message) error
-
-	// TrySend performs a non-blocking send.  Returns error if blocked.
-	TrySend(Message) error
-
 	// Closes the peer connection and the channel returned from Recv().
 	Close()
 
+	// IsLocal returns true if the session is local.
+	IsLocal() bool
+
 	// Recv returns a channel of messages from the peer.
 	Recv() <-chan Message
 
-	// IsLocal returns true if the session is local.
-	IsLocal() bool
+	// Send returns the peer's outgoing message channel.
+	Send() chan<- Message
```

Comment:
I find naming `Send` confusing. It was clear in previous meaning and value. But looking somewhere in the code at line `peer.Send()` it is not clear that it returns a channel now. And as the signature is changed (thus making new design incompatible with old behaviour) - maybe we can give it better name? At least smth like `SendCh`? What do you think?

Reply:
The reason I left it as `Send` is because that pairs with the existing `Recv` function that also returns a channel. If we change `Send` to something else, then we should also change `Recv`.
```go
// Recv returns a channel of messages from the peer.
Recv() <-chan Message
// Send returns the peer's outgoing message channel.
Send() chan<- Message
```

Reply:
Yeah, I see. While we're making a refactoring and can give better naming for both, I'm ok to keep it as is. So it's up to you: if you agree - let's rename! If no - no problem.